### ISSUE TITLE
Imported vector data misaligned when CRS set to 5514

### ISSUE THREAD

**Issue by jmacura** [OpenOrienteering/mapper#1981]
### Steps to reproduce
1. set coordinate system to EPSG:5514 (S-JTSK / Krovak EastNorth, the national system of CZ)
2. add raster template to Mapper
3. File->Import->Anything (tested SHP and GPKG)

### Actual behaviour
The imported data are misaligned by some 13 metres.
By follow-up verification, I have realized that the imported vector data are misplaced while the raster (JPG, tested also with GeoTIFF) is placed correctly.

### Expected behaviour
All data should fit.

### Configuration
Mapper Version: 0.9.5, 0.9.4, 0.9.2
Operating System: Windows 10 Pro 20H2

![image](https://user-images.githubusercontent.com/5598693/127236963-953ae804-f569-451e-ae78-a45f59fc0f64.png)
The semi-transparent-purple areas are buildings imported from Shapefile. Contours (here in blue) are imported from another Shapefile and also shifted. The WMS84->local system transformation is not to blame here as the contours were created in the EPSG:5514 CRS from a point-cloud originally captured in EPSG:5514.

![image](https://user-images.githubusercontent.com/5598693/127236757-e9371c64-195f-435e-a69e-3f2f91e7e110.png)
(More or less) the same area in QGIS 3.20 where everything fits.

![image](https://user-images.githubusercontent.com/5598693/127236990-255a422a-d07c-493a-9e9b-bf02e1c35dca.png)
Now the full-purple areas are the same buildings imported from Shapefile, yet now were saved in EPSG:3045 (ETRS89/UTM33N)

**Comment by krticka** [comment]
Yes, there is a problem with importing 5514 vector data into 5514 project when transformation is adding extra shift to the imported data.  Current workaround is to import these data just using real coordinates only (without using CRS).

**Comment by jmacura** [comment]
> 
> 
> Yes, there is a problem with importing 5514 vector data into 5514 project when transformation is adding extra shift to the imported data. Current workaround is to import these data just using real coordinates only (without using CRS).

Could you be more descriptive about that workaround? Shall I delete the CRS information from the file?

**Comment by krticka** [comment]
Deleting the PRJ file when using shapefile is the easiest solution.

**Comment by jmacura** [comment]
> 
> 
> Deleting the PRJ file when using shapefile is the easiest solution.

Oddly, it indeed works. Díky za tip!

**Comment by jmacura** [comment]
This is a sad story. I have opened this issue some two years ago and meanwhile myself forgot about this issue. Today I have spent 2 hours figuring out why the map background is misaligned against imported contour data and my colleague spent countless hours mapping on misaligned contours 😭 

I am not posting this just to complaint. I am aware OO Mapper is a voluntary project (and a great one!). I am posting this in curiosity whether I am really the only one hitting this issue again and again?

**Comment by dg0yt** [comment]
If I didn't miss it: Did you provide test files to reproduce and resolve the issue? We can establish a confidential channel for sensitive data.

**Comment by lpechacek** [comment]
FWIW, (public) test data for this issue.
[issue-1981-shifted-krovak.zip](https://github.com/OpenOrienteering/mapper/files/11110442/issue-1981-shifted-krovak.zip)
